app/src/main/java/com/vocabulary/myvocabulary/ui/dictionaries/ShareDictionaryViewModel.kt

```diff
@@ -0,0 +1,159 @@
+
+                    for (csvRecord in csvParser) {
+                        insertWordToDatabase(Word(
+                                containerDictionaryId = dictionaryId,
+                                translation = csvRecord.get(0),
+                                word = csvRecord.get(1),
+                                created = Calendar.getInstance().time
+                        ))
+                    }
+                } catch (ex: Exception) {
+                    ex.printStackTrace()
+                }
+            }
+
+        }
+
+    }
+
+    private fun insertWordToDatabase(word: Word) {
+        disposables += Completable.fromCallable { wordRepository.createWord(word) }
+                .subscribeOn(rxSchedulers.io())
+                .observeOn(rxSchedulers.main())
+                .subscribe()
+    }
+
+    private fun observeIsImport() {
+        disposables += shareDictionaryRepository.getImport()
+                .subscribeOn(rxSchedulers.io())
+                .observeOn(rxSchedulers.main())
+                .subscribe{ isImport.postValue(it) }
+    }
+
+    fun setIsImport(isImport: Boolean) {
+        shareDictionaryRepository.setIsImport(isImport)
+    }
+
+    fun saveCsvData(csv: Uri) {
+        shareDictionaryRepository.storeCsvData(csv)
+    }
+
+    private fun writeCsvFile(words: List<Word>, context: Context): File {
+        var fileWriter: FileWriter? = null
+        var csvPrinter: CSVPrinter? = null
+        val file = File("${context.filesDir.path}/export_dictionary.csv")
+        try {
+            file.createNewFile()
+            fileWriter = FileWriter(file)
+            csvPrinter = CSVPrinter(fileWriter, CSVFormat.DEFAULT)
+
+            for (word in words) {
+                val data = Arrays.asList(
+                        word.translation,
+                        word.word
+                )
+                csvPrinter.printRecord(data)
+            }
+        } catch (e: Exception) {
+            e.printStackTrace()
+            Log.e("ERROR", e.message)
+        } finally {
+            try {
+                fileWriter!!.flush()
+                fileWriter.close()
+                csvPrinter!!.close()
+            } catch (e: IOException) {
+                e.printStackTrace()
+                Log.e("ERROR", e.message)
+            }
+        }
+        return file
+    }
+
+    fun shareDictionary(words: List<Word>, context: Context) {
+        val file = writeCsvFile(words, context)
+        val intent = Intent(Intent.ACTION_SEND)
+        intent.type = "text/csv"
+        intent.putExtra(
+                Intent.EXTRA_STREAM,
+                FileProvider.getUriForFile(
+                        context,
+                        context.applicationContext.packageName + ".fileprovider",
+                        file)
+        )
+        intent.addFlags(Intent.FLAG_GRANT_READ_URI_PERMISSION)
+        startActivity(context, Intent.createChooser(intent, "Share file"), null)
```

Comment:
"Share file" should come from the resources.

Reply:
Done